src/persist/src/postgres.rs

```diff
@@ -0,0 +1,252 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Implementation of [Consensus] backed by Postgres.
+
+use std::sync::Arc;
+use std::time::Instant;
+
+use async_trait::async_trait;
+use tokio::sync::Mutex;
+use tokio::task::JoinHandle;
+use tokio_postgres::{Client as PostgresClient, NoTls};
+
+use mz_ore::task;
+
+use crate::error::Error;
+use crate::location::{Consensus, ExternalError, SeqNo, VersionedData};
+
+const SCHEMA: &str = "
+CREATE TABLE IF NOT EXISTS consensus (
+    shard text NOT NULL,
+    sequence_number bigint NOT NULL,
+    data bytea NOT NULL,
+    PRIMARY KEY(shard, sequence_number)
+);
+";
+
+/// Configuration to connect to a Postgres backed implementation of [Consensus].
+#[derive(Clone, Debug)]
+pub struct PostgresConsensusConfig {
+    url: String,
+}
+
+impl PostgresConsensusConfig {
+    const EXTERNAL_TESTS_POSTGRES_URL: &'static str =
+        "MZ_PERSIST_EXTERNAL_STORAGE_TEST_POSTGRES_URL";
```

Comment:
we should figure out how to set this in CI sooner than later. probably also fine as a followup but we should track it in  11605 so it doesn't get dropped

Reply:
I think the way I've wired it up worked? I didn't even need to add postgres to the mz composition I just added a new env var. I don't have a clear artifact of whether i correctly wired things up or not other than - before wiring the postgres consensus test succeeded in 0.00008 seconds, and after wiring the env variable through the test took .28 seconds.

Reply:
i also agree with nikhil. until we hit something that's different between the two for our usage then we should just be testing vs the easy thing

> also sanity check manually that the implementation as written works with aurora.

double checking that you did this and it worked?

Reply:
yup! just tested against aurora and it works!